### ISSUE TITLE
[feature request]  Bash on Ubuntu on Windows (WSL)

### ISSUE THREAD

**Issue by francescoboc** [platformio/platformio-atom-ide-terminal#435]
Would it be possible to load a Bash on Ubuntu on Windows terminal? (WSL)

Thanks

**Comment by HudsonMercer** [comment]
It is already possible however it breaks the arrow keys as well as a few other nav/function keys.

**Comment by francescoboc** [comment]
yes it works, that's great! but the breaking of the arrow keys is quite unconfortable, expecially for navigating into the last commands inserted... is it there a fix for that?

thank you very much,
f.

**Comment by HudsonMercer** [comment]
There is no fix that can be implemented on the users side that I am aware of. It has to do with how bash sends keys to the console in question. It's sending windows keycodes to a unix system and causing it to do basically nothing from what i understand. You can read more about it here. 

https://github.com/Microsoft/WSL/issues/111

People using VScode were running into the same issues when using WSL inside of VScode

**Comment by c-sonntag** [comment]
Hey everybody, I came across your __issue__ by looking for "WSL platformio compatibility ..."
It makes me think, when I use **MSYS2** (http://www.msys2.org/) to fix this kind of problem as with "ghc" (for haskell development), I use the program "winpty".
```bash
# Such as a simple script "ghci.sh":
exec winpty -- ghc --interactive "$@"
```
with `ghc --interactive "$@"` as a command for "winpty".
Maybe, if the "winpty" program exists for **WSL** it can help you.

**Comment by henryxparker** [comment]
Associated to #221 and #231

**Comment by mslinn** [comment]
You might want to know about the new Windows Terminal API and console. This project is a major effort by Microsoft, moving forward with high visibility
  * https://github.com/microsoft/terminal
  * https://devblogs.microsoft.com/commandline/windows-command-line-introducing-the-windows-pseudo-console-conpty/

**Comment by the-j0k3r** [comment]
@mslinn Ive hidden your comment as offtopic and just to be clear, if you or anyone else has reports about platformio-ide-terminal working or not working with WSL please feel free to post your findings in detail here.

Related to WSL issues with platformio-ide-terminal is also issue #235. Please familiarize yourselfs with the reported issues on these two tickets and proceed ontopic.

Thanks for your understanding.

**Comment by mslinn** [comment]
Your comment seems to suggest that you believe the new Windows API for consoles and the new WIndows Terminal is WSL-specific. It is not. Windows Terminal and the API it is built from is applicable to all of Windows, and in fact is orthogonal to WSL and does not require WSL.

**Comment by the-j0k3r** [comment]
This topic/issue is not related to the new Windows API for consoles or new Terminal.

Thanks for your understanding on this.

**Comment by einarpersson** [comment]
+1 for WSL support. **WSL2** will be a game changer. If I'd understod it correctly USB/serial will not be supported in WSL2 from day 1 but probably in a near future. 

_Reeaally_ hope that PlatformIO will work with WSL2 in a not too distant future
